pandas/core/generic.py

```diff
@@ -6393,6 +6398,19 @@ def replace(
         regex=False,
         method="pad",
     ):
+        if not (
+            is_scalar(to_replace)
+            or isinstance(to_replace, pd.Series)
+            or is_re_compilable(to_replace)
+            or is_list_like(to_replace)
+            or is_dict_like(to_replace)
+        ):
+            raise TypeError(
+                f"Expecting 'to_replace' to be str, regex, list, dict, Series, "
```

Comment:
```suggestion
                "Expecting 'to_replace' to be str, regex, list, dict, Series, "
```

Minor but this isn't necessary